tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -678,6 +680,12 @@ private static void setTraceRequestId(Optional<String> requestId) {
         }
     }
 
+    private static void logSettingTrace(Trace trace) {
+        if (log.isDebugEnabled()) {
+            log.debug("Setting trace", SafeArg.of("trace", trace));
```

Comment:
nit: the trace object itself probably won't tell us a lot, I think we mostly care about traceId and isObservable, but those should exist in MDC.

If we remove the non-constant args, we no longer need an isDebugEnabled check
```suggestion
            log.debug("Setting trace");
```